pkg/parse/root.go

```diff
@@ -396,14 +396,22 @@ func (p *root) setSyncStatusWithRetries(ctx context.Context, newStatus syncStatu
 
 	setSyncStatusFields(&rs.Status.Status, newStatus, denominator)
 
-	errorSources, errorSummary := summarizeErrors(rs.Status.Source, rs.Status.Sync)
-	if newStatus.syncing {
-		rootsync.SetSyncing(rs, true, "Sync", "Syncing", rs.Status.Sync.Commit, errorSources, errorSummary, rs.Status.Sync.LastUpdate)
-	} else {
-		if errorSummary.TotalCount == 0 {
-			rs.Status.LastSyncedCommit = rs.Status.Sync.Commit
+	// The Syncing condition should only represent the status and errors for the latest commit.
+	// So only update the Syncing condition here if we haven't fetched a new commit.
+	// Ideally, checking the source commit would be enough, but because fetching and parsing share the source status,
+	// we also have to check the rendering commit, which may be updated first.
+	var lastSyncStatus string
+	if rs.Status.Source.Commit == rs.Status.Sync.Commit && rs.Status.Rendering.Commit == rs.Status.Sync.Commit {
```

Comment:
Correct me if I'm wrong, but I think the intent is that the `Syncing` condition should be set to `true` here whenever the applier is currently running (`newStatus.syncing`).

When the applier is running it's expected that the on-cluster RSync will have updated `status.source` and `status.rendering`, but not yet `status.sync`. The `status.sync` field is set by `setSyncStatusFields` call above, which won't be written to the cluster until the Update below on line 440.

I believe this means the first time the applier calls [SetSyncStatus](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/parse/run.go#L613), the `rs.Status.Source.Commit` won't yet be updated and we would skip the condition update here. To avoid this I think this comparison needs to compare the commits for `currentRS` instead of `rs`.

TL;DR: `s/rs/currentRS/g`

Reply:
`rs.Status.Source.Commit` updates are published by `setSourceStatus`, not `SetSyncStatus`. So that's always published to the cluster before `SetSyncStatus` runs.

`rs.Status.Sync.Commit` is updated locally by `setSyncStatusFields`, so it's safe to use it before it's published to the cluster.

Reply:
Ah I mixed up the two, `currentRS` is the unchanged on-cluster RSync and `rs` is the updated one. Disregard